python/arcticdb/version_store/library.py

```
@@ -1261,6 +1261,77 @@ def reload_symbol_list(self):
         """
         self._nvs.version_store.reload_symbol_list()
 
+    def is_symbol_fragmented(self, symbol: str, segment_size: int = None) -> bool:
+        """
+        Check whether the number of segments that would be reduced by compaction is more than or equal to the
+        value specified by the configuration option "SymbolDataCompact.SegmentCount" (defaults to 100).
+
+        Parameters
+        ----------
+        symbol: `str`
+            Symbol name.
+        segment_size: `int`
+            Target for maximum no. of rows per segment, after compaction.
+            If parameter is not provided, library option for segments's maximum row size will be used
+
+        Notes
+        ----------
+        Config map setting - SymbolDataCompact.SegmentCount will be replaced by a library setting
+        in the future. This API will allow overriding the setting as well.
+        
+        Returns
+        -------
+        bool
+        """
+        return self._nvs.is_symbol_fragmented(symbol, segment_size)
```

This should raise an exception with an error code!